Doc/conf.py

```diff
@@ -319,35 +322,41 @@
     'collapsiblesidebar': True,
     'issues_url': '/bugs.html',
     'license_url': '/license.html',
-    'root_include_title': False   # We use the version switcher instead.
+    'root_include_title': False,  # We use the version switcher instead.
 }
 
-if os.getenv("READTHEDOCS"):
-    html_theme_options["hosted_on"] = '<a href="https://about.readthedocs.com/">Read the Docs</a>'
+if os.getenv('READTHEDOCS'):
+    html_theme_options['hosted_on'] = (
+        '<a href="https://about.readthedocs.com/">Read the Docs</a>'
+    )
 
 # Override stylesheet fingerprinting for Windows CHM htmlhelp to fix GH-91207
 # https://github.com/python/cpython/issues/91207
 if any('htmlhelp' in arg for arg in sys.argv):
     html_style = 'pydoctheme.css'
-    print("\nWARNING: Windows CHM Help is no longer supported.")
-    print("It may be removed in the future\n")
+    print('\nWARNING: Windows CHM Help is no longer supported.')
+    print('It may be removed in the future\n')
 
 # Short title used e.g. for <title> HTML tags.
 html_short_title = f'{release} Documentation'
 
 # Deployment preview information
 # (See .readthedocs.yml and https://docs.readthedocs.io/en/stable/reference/environment-variables.html)
-repository_url = os.getenv("READTHEDOCS_GIT_CLONE_URL")
+is_deployment_preview = os.getenv('READTHEDOCS_VERSION_TYPE') == 'external'
+repository_url = os.getenv('READTHEDOCS_GIT_CLONE_URL', '')
+repository_url = repository_url.removesuffix('.git')
 html_context = {
-    "is_deployment_preview": os.getenv("READTHEDOCS_VERSION_TYPE") == "external",
-    "repository_url": repository_url.removesuffix(".git") if repository_url else None,
-    "pr_id": os.getenv("READTHEDOCS_VERSION"),
-    "enable_analytics": os.getenv("PYTHON_DOCS_ENABLE_ANALYTICS"),
+    'is_deployment_preview': is_deployment_preview,
+    'repository_url': repository_url or None,
+    'pr_id': os.getenv('READTHEDOCS_VERSION'),
+    'enable_analytics': os.getenv('PYTHON_DOCS_ENABLE_ANALYTICS'),
```

Comment:
I think a bunch of these changes would now no longer be applied if you did a clean autoformat on the directory as it exists on `main`, now we've switched to `quote-style = "preserve"`. It looks like there may be a few changes you've applied manually here as well as ones automatically applied by Ruff... but I'd consider reverting all changes to `.py` files currently made in this PR, then re-running `pre-commit run -a` with the new config settings